plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TableSnapshot.java

```diff
@@ -253,23 +259,46 @@ private MetadataEntry getCheckpointMetadataEntry(
             LastCheckpoint checkpoint)
             throws IOException
     {
+        MetadataEntry metadata = null;
+        ProtocolEntry protocol = null;
         for (Location checkpointPath : getCheckpointPartPaths(checkpoint)) {
             TrinoInputFile checkpointFile = fileSystem.newInputFile(checkpointPath);
-            Stream<DeltaLakeTransactionLogEntry> metadataEntries = getCheckpointTransactionLogEntries(
+            Iterator<DeltaLakeTransactionLogEntry> entries = getCheckpointTransactionLogEntries(
                     session,
-                    ImmutableSet.of(METADATA),
+                    ImmutableSet.of(METADATA, PROTOCOL),
```

Comment:
This change likely breaks the checkpoint reading optimizations added in https://github.com/trinodb/trino/commit/06e1339755251675a8104d523be3461b55211dd7

We should probably adopt the optimization to support or-ing multiple tuple domains.

Reply:
Please file an issue to https://github.com/trinodb/trino/issues/new